### PR TITLE
Order history

### PR DESCRIPTION
**Pull request by NgocAnhDT**
## Related Tickets:
- [#46233](https://edu-redmine.sun-asterisk.vn/projects/hn_oe49-shoe-store/issues/46233)

## WHAT this PR does?
- Order History
- Fix edit product

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Review by ariesh2t — APPROVED** [review]

**Comment by NgocAnhDT** [comment]
ready

File: resources/views/users/history/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                    @foreach ($orders as $order)
+                        @php
+                            $numOfProduct = 0;
+                            foreach ($order->products as $product) {
+                                $numOfProduct++;
+                            }
+                        @endphp
+                        <tr>
+                            <td class="text-center">{{ $order->id }}</td>
+                            <td class="text-center">{{ $numOfProduct }}</td>
```
Comment (manhtd-1083) [inline resources/views/users/history/index.blade.php:33]:
```suggestion
                        <tr>
                            <td class="text-center">{{ $order->id }}</td>
                            <td class="text-center">{{ $order->products->count() }}</td>
```

File: resources/views/users/history/show.blade.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                            <th>{{ __('product_name') }}</th>
+                            <th class="w-10">{{ __('price') }}</th>
+                            <th class="w-13">{{ __('quantity') }}</th>
+                            <th class="w-15">{{ __('total price') }}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        @php $total_price = 0; @endphp
+                        @foreach ($orders->products as $product)
+                            @php $total_price += $product->price * $product->pivot->quantity @endphp
```
Comment (manhtd-1083) [inline resources/views/users/history/show.blade.php:45]: cái xử lý này đáp sang bên controller đc ko? ta nên tránh để code logic bên view ntn nha

**Comment by NgocAnhDT** [comment]
ready

**Pull request merged**